internal/cmd/agent/deployer/normalizers/status.go

```diff
@@ -0,0 +1,13 @@
+package normalizers
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
```

Comment:
Feels like we could write a test for this code here?

Also, I wonder about something like this in `internal/cmd/agent/deployer/internal/diff/diff.go` which would simplify this (and the noopNormalizer).


```go
// The NormalizerFunc type is an adapter to allow the use of ordinary functions
// as Normalizers. If f is a function with the appropriate signature,
// NormalizerFunc(f) is a Normalizer that calls f.
type NormalizerFunc func(*unstructured.Unstructured) error

// Normalize calls f(un).
func (f NormalizerFunc) Normalize(un *unstructured.Unstructured) error {
    return f(un)
}
```

```go
// GetNoopNormalizer returns normalizer that does not apply any resource modifications
func GetNoopNormalizer() Normalizer {
    return NormalizerFunc(func(*unstructured.Unstructured) error {
        return nil 
    })  
}
```

Which could even be simplified further to...

```go
// NoopNormalizer is a normalizer that does not apply any resource modifications
var NoopNormalizer = NormalizerFunc(func(*unstructured.Unstructured) error {
    return nil 
})  
```

Where the StatusNormalizer would become...

```go
// StatusNormalizer removes a top-level "status" fields from the object, if present
func NormalizeStatus(un *unstructured.Unstructured) error {
    unstructured.RemoveNestedField(un.Object, "status")
    return nil 
}
```

Reply:
While I agree with you, I believe such refactoring belongs to a separate PR. I prefer keeping things as independent as possible.

To be honest, I didn't even know these code existed before investigating this issue, so maybe the same happens for other team members and they would like to review it more deeply and apply any needed refactorings.

About adding a test, I didn't do it initially because I thought it wouldn't provide much value, since it would be like testing `RemoveNestedField` instead, but I'll add it since it doesn't hurt either and should be easy. Thanks.